File: local_examples/cmds_cnxmgmt/go-redis/cmds_cnxmgmt_test.go
// EXAMPLE: cmds_cnxmgmt
// HIDE_START
package example_commands_test

import (
	"context"
	"fmt"

	"github.com/redis/go-redis/v9"
)

func ExampleClient_cnxmgmt_auth() {
	ctx := context.Background()

	rdb := redis.NewClient(&redis.Options{
		Addr:     "localhost:6379",
		Password: "", // no password docs
		DB:       0,  // use default DB
	})

	// AUTH is connection-scoped, so run it on a dedicated connection from
	// Conn() rather than on the pooled client.
	conn := rdb.Conn()
	defer conn.Close()
	// HIDE_END

	// STEP_START auth1
	authResult1, err := conn.Auth(ctx, "temp_pass").Result()

	if err != nil {
		fmt.Println(err)
	}

	fmt.Println(authResult1) // >>> OK

	authResult2, err := conn.AuthACL(ctx, "default", "temp_pass").Result()

	if err != nil {
		fmt.Println(err)
	}

	fmt.Println(authResult2) // >>> OK
	// STEP_END

	// STEP_START auth2
	authResult3, err := conn.AuthACL(ctx, "test-user", "strong_password").Result()

	if err != nil {
		fmt.Println(err)
	}

	fmt.Println(authResult3) // >>> OK
	// STEP_END
	// HIDE_START
}

// HIDE_END

File: local_examples/cmds_cnxmgmt/predis/CmdsCnxmgmtTest.php
// EXAMPLE: cmds_cnxmgmt
<?php
use Predis\Client as PredisClient;

class CmdsCnxmgmtTest
// REMOVE_START
extends PredisTestCase
// REMOVE_END
{
    public function testCmdsCnxmgmt() {
        $r = new PredisClient([
            'scheme'   => 'tcp',
            'host'     => '127.0.0.1',
            'port'     => 6379,
            'password' => '',
            'database' => 0,
        ]);

        // STEP_START auth1
        $res1 = $r->auth('temp_pass');
        echo $res1 . PHP_EOL; // >>> OK

        $res2 = $r->auth('default', 'temp_pass');
        echo $res2 . PHP_EOL; // >>> OK
        // STEP_END

        // STEP_START auth2
        $res3 = $r->auth('test-user', 'strong_password');
        echo $res3 . PHP_EOL; // >>> OK
        // STEP_END
    }
}
